src/main/java/gov/usgs/owi/nldi/controllers/LinkedDataControllerV2.java

```diff
@@ -0,0 +1,124 @@
+		@RequestParam(value=Parameters.STOP_COMID, required=false) @Range(min=1, max=Integer.MAX_VALUE) String stopComid,
+		@Parameter(description=Parameters.DISTANCE_DESCRIPTION)
+		@RequestParam(value=Parameters.DISTANCE, required=false, defaultValue=Parameters.MAX_DISTANCE)
+		@Pattern(message=Parameters.DISTANCE_VALIDATION_MESSAGE, regexp=Parameters.DISTANCE_VALIDATION_REGEX) String distance,
+		@RequestParam(value=Parameters.LEGACY, required=false) String legacy) throws Exception {
+
+		BigInteger logId = logService.logRequest(request);
+
+		try {
+			List<Map<String, Object>> dataSources = controllerV1.getDataSources(request, response);
+			List<Map<String, Object>> newDataSources = new ArrayList<>();
+			String newNavigationUrl = createNewNavigationUrl(request);
+			for (Map<String, Object> dataSource: dataSources) {
+				if ("comid".equals(dataSource.get("source"))) {
+					dataSource.put("source", "Flowlines");
+					dataSource.put("sourceName", "NHDPlus flowlines");
+					String flowlinesV2Url = newNavigationUrl.replace("linked-data/", "linked-data/v2/");
+					dataSource.put("features", flowlinesV2Url  + "flowlines");
+					newDataSources.add(dataSource);
+				} else {
+					dataSource.put("features", newNavigationUrl
+						+ dataSource.get("source").toString().toLowerCase());
+					newDataSources.add(dataSource);
+				}
+			}
+			return newDataSources;
+
+		} catch (Exception e) {
+			GlobalDefaultExceptionHandler.handleError(e, response);
+		} finally {
+			logService.logRequestComplete(logId, response.getStatus());
+		}
+		return null;
+	}
+
+	//swagger documentation for /linked-data/{featureSource}/{featureID}/navigate/{navigationMode} endpoint
+	@Operation(summary = "getFlowlines", description = "returns the flowlines for the specified navigation in WGS84 lat/lon GeoJSON")
+	@GetMapping(value="linked-data/v2/{featureSource}/{featureID}/navigate/{navigationMode}/flowlines", produces=MediaType.APPLICATION_JSON_VALUE)
+	public void getFlowlines(
+		HttpServletRequest request, HttpServletResponse response,
+		@PathVariable(LookupDao.FEATURE_SOURCE) String featureSource,
+		@PathVariable(Parameters.FEATURE_ID) String featureID,
+		@PathVariable(Parameters.NAVIGATION_MODE) @Pattern(regexp=REGEX_NAVIGATION_MODE) String navigationMode,
+		@RequestParam(value=Parameters.STOP_COMID, required=false) @Range(min=1, max=Integer.MAX_VALUE) String stopComid,
+		@Parameter(description=Parameters.DISTANCE_DESCRIPTION)
+		@RequestParam(value=Parameters.DISTANCE, required=false, defaultValue=Parameters.MAX_DISTANCE)
+		@Pattern(message=Parameters.DISTANCE_VALIDATION_MESSAGE, regexp=Parameters.DISTANCE_VALIDATION_REGEX) String distance,
+		@RequestParam(value=Parameters.LEGACY, required=false) String legacy) throws Exception {
+
+		BigInteger logId = logService.logRequest(request);
+
+		try {
+			controllerV1.getFlowlines(request, response, featureSource, featureID, navigationMode, stopComid, distance, legacy);
```

Comment:
Typically, controllers should stay separate and not have inter dependencies. Understand, the desire to not to duplicate code. In this case, since the code is small, it would be ok to have the controller logic  here as well.
If the V1 version is really deprecated, controllerV1.getFlowlines() should not really get updated. If the code is here, the v1 controller can be deleted with out impacting this controller.

Reply:
Hmmm.  My original thought was to have the code self-documenting, as in "right now this v2 api is doing the exact same as a different v1 endpoint), but probably you are right.  I will change and add a comment instead.

Reply:
Actually, I went to implement this and there was a snag.   I can copy a few lines from getFlowlines(), no problem.   But I also need getDataSources(), and if I copy that over, I need to copy over all the tests for getDataSources() as well.  And then we have quite a bit of duplicated code.

So, I'm going to copy over the code for getFlowlines() because it is deprecated but leave the dependency on controllerV1 for getDataSources().   Let me know what you think.   

I have seen this pattern of controllers invoking other controllers before.  I worked on a large project where multiple teams owned controllers and it was sometimes necessary to have your ABCController invoke someone's XYZController.  Also, for versioning, that project used the same approach I'm using here.  Which is not say to that was the correct way to do it, only that it did work and I don't recall it creating any problems or confusion.

Reply:
I think that is reasonable. This should leave no code dependent on the deprecated method.